### PR TITLE
Upgrade ember and add hooks for testing

### PR DESCRIPTION
**Pull request by juwara0**
#MINOR#

Closes: https://github.com/ciena-frost/ember-frost-info-bar/issues/33

# CHANGELOG

* **Updated** Ember, Ember-CLI and Ember data to ~2.6.0
* **Updated** npm and bower dependencies
* **Added** `ember-prop-types` to the frost-info-bar component
* **Added** a default hook name for the frost-info-bar component
* **Added** hook names for different block-slot sections in the frost-info-bar component
* **Added** tests to validate the new hook names in the frost-info-bar component
* **Updated** README to explain usage of the hooks for the frost-info-bar component

### CODE REVIEW

File: blueprints/ember-frost-info-bar/index.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ module.exports = {
     return this.addAddonsToProject({
       packages: [
         {name: 'ember-frost-core', target: '>=0.8.0 <2.0.0'},
-        {name: 'ember-block-slots', target: '>=0.12.1", <1.0.0'}
+        {name: 'ember-block-slots', target: '>=0.12.1", <1.0.0'},
```
Comment (sglanzer) [inline blueprints/ember-frost-info-bar/index.js]: This should be moved to `>=1.0.0, <2.0.0`

File: package.json [hunk 2]
```diff
@@ -18,30 +18,37 @@
...
+    "Matthew Dahl (https://github.com/sandersky)",
+    "Quincy Le (https://github.com/quincyle)",
+    "Olufemi Adeojo (https://github.com/oadeojo)",
+    "Gosia Hyndman (https://github.com/vesper2000)",
+    "Michael Carroll (https://github.com/juwara0)"
+  ],
   "license": "MIT",
   "devDependencies": {
     "broccoli-asset-rev": "^2.4.2",
     "ember-block-slots": "^1.1.0",
```
Comment (sglanzer) [inline package.json]: Can you move this to dependencies?

File: package.json [hunk 3]
```diff
@@ -18,30 +18,37 @@
...
     "ember-cli-notifications": "3.2.0",
+    "ember-cli-release": "0.2.9",
+    "ember-cli-sri": "^2.1.0",
     "ember-cli-uglify": "1.2.0",
     "ember-computed-decorators": "0.2.2",
-    "ember-data": "^2.4.0",
+    "ember-data": "~2.6.0",
     "ember-disable-proxy-controllers": "1.0.1",
     "ember-export-application-global": "1.0.5",
     "ember-frost-core": ">=0.20.0 <1.0.0",
```
Comment (sglanzer) [inline package.json]: Same here, dependencies - and we should adjust the range in the blueprint to match `<1.0.0`

File: tests/dummy/app/pods/demo/template.hbs [hunk 4]
```diff
@@ -17,7 +17,7 @@
       &lt;placeholder: controls&gt;
   {{/block-slot}}
   {{#block-slot 'actions' as |action|}}
-    {{action.button icon='frost/infobar-create' text='Create' onClick=(action 'createUser')}}
+    {{action.button icon='frost/infobar-create' text='Click me!' onClick=(action 'triggerAction')}}
```
Comment (sglanzer) [inline tests/dummy/app/pods/demo/template.hbs]: Very minor, but I'd like to start consistently line-wrapping the properties - e.g.

```
{{action.button
  icon='infobar-create'
  text='Click me!'
  onClick=(action 'triggerAction')
}}
```

Comment (sglanzer) [inline tests/dummy/app/pods/demo/template.hbs]: We should also make a push to eliminate references to nested icon paths - i.e. `frost/infobar-create` should be `infobar-create` - although I'm curious why we have a second icon for a `plus`

**Comment by sglanzer** [comment]
Approved - good stuff, thanks for taking the initiative on this

It's not really part of this PR, but we should probably fix the dependency issues before merging

File: blueprints/ember-frost-info-bar/index.js (hunk 1 above)
Comment (juwara0) [inline blueprints/ember-frost-info-bar/index.js]: This has been done.

File: package.json (hunk 2 above)
Comment (juwara0) [inline package.json]: This has been done.

File: package.json (hunk 3 above)
Comment (juwara0) [inline package.json]: This has been done.

File: tests/dummy/app/pods/demo/template.hbs (hunk 4 above)
Comment (juwara0) [inline tests/dummy/app/pods/demo/template.hbs]: These have been done as well.

**Comment by juwara0** [comment]
👍 Self approving after completing the requests listed above.

**Comment by sglanzer** [comment]
Thanks @juwara0!

**Comment by juwara0** [comment]
You are welcome @sglanzer!

**Pull request merged**